src/Symfony/Bundle/SecurityBundle/EventListener/FirewallListener.php

```diff
@@ -30,6 +30,8 @@ class FirewallListener extends Firewall
 
     public function __construct(FirewallMapInterface $map, EventDispatcherInterface $dispatcher, LogoutUrlGenerator $logoutUrlGenerator)
     {
+        // the type-hint will be updated to the "EventDispatcherInterface" from symfony/contracts in 5.0
```

Comment:
Is there a special reason not to do that now?

Reply:
yes: we must pass an EDI from the component because the instance is passed fo Firewall, which calls ExceptionListener::register(), which itself calls addListener, which is a method that doesn't exist in the new interface.